LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DefaultClassCDevicesMessageSender.cs

```diff
@@ -112,19 +112,36 @@ public async Task<bool> SendAsync(IReceivedLoRaCloudToDeviceMessage message, Can
                 fcntDown,
                 this.logger);
 
+            var messageIdLog = message.MessageId ?? "undefined";
+
             if (downlinkMessageBuilderResp.IsMessageTooLong)
             {
                 this.c2dMessageTooLong?.Add(1);
-                this.logger.LogError($"[class-c] cloud to device message too large, rejecting. Id: {message.MessageId ?? "undefined"}");
+                this.logger.LogError($"[class-c] cloud to device message too large, rejecting. Id: {messageIdLog}");
                 if (!await message.RejectAsync())
                 {
-                    this.logger.LogError($"[class-c] failed to reject. Id: {message.MessageId ?? "undefined"}");
+                    this.logger.LogError($"[class-c] failed to reject. Id: {messageIdLog}");
                 }
                 return false;
             }
             else
             {
-                await this.packetForwarder.SendDownstreamAsync(downlinkMessageBuilderResp.DownlinkMessage);
+                if (await this.packetForwarder.SendDownstreamAsync(downlinkMessageBuilderResp.DownlinkMessage))
```

Comment:
I think we should let the exception bubble up and Abandon on exceptions + rethrow and complete otherwise.